crates/ra_hir_def/src/item_scope.rs

```diff
@@ -151,13 +151,20 @@ impl ItemScope {
 }
 
 impl PerNs {
-    pub(crate) fn from_def(def: ModuleDefId, v: Visibility) -> PerNs {
+    pub(crate) fn from_def(def: ModuleDefId, v: Visibility, has_constructor: bool) -> PerNs {
```

Comment:
That's a somewhat iffy API, because we have a boolean parameter *and* it only applies if `def` is of specific types, but I guess we can live with it. 

r=me